File: Swift-C++-Testsuite/test/generics/Swift/ExtensionStack.swift
@main
struct MyMain {

    static func main() {
        var stackOfStrings = Stack<String>()
        stackOfStrings.push("uno")
        stackOfStrings.push("dos")
        stackOfStrings.push("tres")
        stackOfStrings.push("cuatro")
        var stackOfNumbers = Stack<Int>()
        stackOfNumbers.push(1)
        stackOfNumbers.push(2)
        stackOfNumbers.push(3)
        stackOfNumbers.push(4)
        if let topItem = stackOfStrings.topItem {
            print("The top item on the stack is \(topItem).")
        }
        if let topItem = stackOfNumbers.topItem {
            print("The top item on the stack is \(topItem).")
        }
    }
}

struct Stack<Element> {
    var items: [Element] = []
    mutating func push(_ item: Element) {
        items.append(item)
    }
    mutating func pop() -> Element {
        return items.removeLast()
    }
}

extension Stack {
    var topItem: Element? {
        return items.isEmpty ? nil : items[items.count - 1]
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/ExtensionStack.cpp


#include <iostream>
#include <stack>
using namespace std;

struct MyMain
{
    static void main()
    {
        stack<string> stackOfStrings;
        stackOfStrings.push("uno");
        stackOfStrings.push("dos");
        stackOfStrings.push("tres");
        stackOfStrings.push("cuatro");
        stack<int> stackOfNumbers;
        stackOfNumbers.push(1);
        stackOfNumbers.push(2);
        stackOfNumbers.push(3);
        stackOfNumbers.push(4);

        if (!stackOfStrings.empty())
        {
            cout << "The top item on the stack is " << stackOfStrings.top() << endl;
        }
        if (!stackOfNumbers.empty())
        {
            cout << "The top item on the stack is " << stackOfNumbers.top() << endl;
        }
    }
};
int main(){
  MyMain::main();
}